cmds/mount/mount.go

```diff
@@ -48,7 +74,13 @@ func main() {
 	if *ro {
 		flags |= unix.MS_RDONLY
 	}
-	if err := unix.Mount(a[0], a[1], *fsType, flags, strings.Join(data, ",")); err != nil {
-		log.Fatalf("Mount :%s: on :%s: type :%s: flags %x: %v\n", dev, path, *fsType, flags, err)
+	if *fsType == "" {
+		// mandatory parameter for the moment
+		log.Printf("No file system type provided!\n")
+		log.Fatalf("Usage: mount [-r] [-o mount options] -t fstype dev path")
+	} else if err := unix.Mount(a[0], a[1], *fsType, flags, strings.Join(data, ",")); err != nil {
+		log.Printf("Mount :%s: on :%s: type :%s: flags %x: %v\n", dev, path, *fsType, flags, err)
+		informIfUnknownFS(*fsType)
```

Comment:
Why not move this up to before you try to mount?

Reply:
Because I prefer the no error case to be the fastest possible. Boot time is important to me . I assume most mounts will be correct and coming from batch files rather than interactive. In both cases the user will have informative message, but be doing the checks afterwards I give a change to run faster in the normal no error case.